src/sophys/common/devices/pimega.py

```diff
@@ -126,6 +134,35 @@ class Pimega(SingleTrigger, PimegaDetector):
     def __init__(self, name, prefix, **kwargs):
         super(Pimega, self).__init__(prefix, name=name, **kwargs)
 
+    def stage(self):
+        # Make sure the current acquisition status is 'Done'
+        self._acquisition_signal.set(0).wait(timeout=30.0)
+
+        self._acquisition_signal.subscribe(
+            self._acquire_setpoint_changed, EpicsSignal.SUB_SETPOINT
+        )
+
+        return super().stage()
+
+    def unstage(self):
+        super().unstage()
+
+        self._acquisition_signal.unsubscribe(self._acquire_setpoint_changed)
+
+    def _acquire_setpoint_changed(self, value, severity, **kwargs):
+        if self._status is None or self._status.done:
+            return
+
+        if value == 1 and severity == AlarmSeverity.INVALID:
+            exc_messages = (
+                "An alarm has been raised by the IOC, with the following status:",
+                f"IOC: {self.cam.ioc_status_message.get()}",
+                f"Backend: {self.cam.backend_status_message.get()}",
```

Comment:
"following status messages"

status is technically a property of EPICS alarms, so if we want to avoid any ambiguity, we could be slightly more specific

this is very much a nit, feel free to ignore it.